cmd/file_lint.go

```diff
@@ -147,19 +171,27 @@ func executeLint(cmd *cobra.Command, _ []string) error {
 func newLintCmd() *cobra.Command {
 	lintCmd := &cobra.Command{
 		Use:   "lint [flags] lint-file",
-		Short: "Short description here",
-		Long:  `Long description Here`,
-		RunE:  executeLint,
+		Short: "Lint a file against a ruleset",
+		Long: "Lint a file against a ruleset. \n" +
```

Comment:
@mheap @rspurgeon can you please validate these?

Reply:
I'm still not sure what `lint-file` represents in this usage.  I think we're using flags for both the state file and the rules file, so what is `lint-file`?

As far as help text:
```suggestion
		Long: "Validate a decK state file against a linting ruleset, reporting any violations or failures. Report output can be returned in JSON, YAML, or human readable format (see --format). \n" +
```

Reply:
I also think it's important we document the `--fail-severity` flag and how it affects the return code of the command. I assume that users of this will want to branch behaviors based on command return code.   I think we should provide details here.  I would make a suggestion, but I'm not sure at this point how the code works in that regard.

Reply:
Thanks for the suggestion!

Fixed this. https://github.com/Kong/deck/pull/981/commits/3bdd2fe76030a629dea7e41ca3b1cc94942219c1

Reply:
> I would make a suggestion, but I'm not sure at this point how the code works in that regard.
Basically `--fail-severity` will gate what returns a non-zero (1) exit code.

Also, is `--display-only-failures` expected to care about `error` only or it should adapt to the `--fail-severity` value? If  the `--display-only-failures` flag should care only about `error` I suggest we change it to `--display-only-errors`.

Thoughts? @rspurgeon @mheap

Reply:
`--fail-severity` and `--display-only-failures` are taken from Spectral, which is the defacto implementation of linting https://docs.stoplight.io/docs/spectral/9ffa04e052cc1-spectral-cli

We should adapt to the `--fail-severity` value

From Spectral:

> -D, --display-only-failures only output results equal to or greater than --fail-severity [boolean] [default: false]

Reply:
@mheap thanks for the clarifying!

https://github.com/Kong/deck/pull/981/commits/a4e025cff686d8ec0e68422901e621c141a7cbc7

Reply:
Can we just adopt what the spectral CLI supports directly?  Can we update our usage to be:
```
-F, --fail-severity results of this level or above will trigger a failure exit code
[string] [choices: "error", "warn", "info", "hint"] [default: "error"]
-D, --display-only-failures only output results equal to or greater than --fail-severity [boolean] [default: false]
```